### ISSUE TITLE
iOS Error crashes game (7.5/8.0) 

### ISSUE THREAD

**Issue by shawna-p** [renpy/renpy#4134]
I'm looking to release an iOS build of an app before Christmas and cannot successfully run the iOS build after installation. The game will build and install fine, but it gets stuck on the splash screen and crashes with the following error:

> Exception = (NSException *) "Failed To call eglInitialize()"

Xcode opens main.c when displaying this error, pointing to the `main` function. 

I downloaded both 8.0.3 and 7.5.3 to see if it made any difference and also tried building The Question, to no success and the same error message. However, I was able to make a functioning build in 7.4.11 with the same game (just downgraded). 

Also perhaps notable are the following deprecations I get warnings about:

'UIActivityIndicatorViewStyleWhiteLarge" is deprecated
'keyWindow' is deprecated
Option -no_dead_strip_inits_and_terms is obsolete

**Comment by renpytom** [comment]
One thing to try here would be to generate a new iOS project for your game, rather than re-using the same project used for 7.4. The iOS projects need to be updated when Ren'Py is, and the customization moved over. 

Please also try the 2022-12-09 nightly from https://nightly.renpy.org , when it's out. I was able to run a game in the iOS simulator with it. (Waiting for my iPad to finish updating so I can test.)

**Comment by shawna-p** [comment]
Ah, perhaps it was unclear - I tried building on 8.0.3, 7.5.3, and 7.4.11 with fresh projects, and tried getting rid of the Xcode project and rebuilding it each time as well. Specifically I tried building the game I want to release on iOS first in 8.0 and got this issue, so then I moved it to 7.5 and created a new Xcode project for it and ran into the same issue. Then I tried building The Question and got the same issue until I moved down to 7.4.11, where it wouldn't crash on opening. So there was 100% no reuse of older Xcode projects from older Ren'Py to newer Ren'Py versions, as I tried the newer versions before I downgraded to try getting a build that worked.

I'm away from my build computer but I'll grab the nightly when I get a chance. Thanks!
